### PR TITLE
Stop command with option to force-terminate

### PR DESCRIPTION
**Pull request by shriramshankar**

### CODE REVIEW

File: src/a/refocus-collector-stop.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
+/**
```
Comment (shriramshankar) [inline src/a/refocus-collector-stop.js]: This was checked in by mistake. Will remove this file tomorrow.

File: test/utils/queueUtils.js [hunk 2]
```diff
@@ -82,13 +93,44 @@ describe('test/utils/queueUtils.js - queue utils unit tests >', () => {
     return done();
   });
 
+  it('flush all buffered queue objects', (done) => {
```
Comment (annyhe) [inline test/utils/queueUtils.js]: these tests mean my PR for the relatedLinks will have to wait.  We can postpone that PR until this one gets merged.

File: src/commands/index.js [hunk 3]
```diff
@@ -37,7 +36,7 @@ program.on('--help', () => {
   console.log('    $ refocus-collector --help');
   console.log('    $ refocus-collector start --collectorName=test ' +
     '--refocusUrl=https://refocus.abczyx.com --accessToken=eygduyguygijfdhkfjhkfdhg');
-  console.log('    $ refocus-collector stop --collectorName=test');
+  console.log('    $ refocus-collector1 stop --collectorName=test');
```
Comment (annyhe) [inline src/commands/index.js]: are we explicitly changing refocus-collector to refocus-collector1?

Comment (shriramshankar) [reply src/commands/index.js]: Will fix it. we are not changing the command name.

**Comment by annyhe** [comment]
@shriramshankar please ask @jgraff2 to review this PR. My schedule is quite full and don't want to hold up this PR.

File: test/commands/stop.js [hunk 4]
```diff
@@ -11,22 +11,183 @@
...
+
+  it('force stop, flush should not be called', (done) => {
+    start.execute(collectorName, refocusUrl, accessToken,
+      { refocusProxy, dataSourceProxy }
+    )
+    .then(() => {
+      // make sure start created the repeat
+      expect(repeater.tracker.heartbeat).to.be.an('object');
+    })
+    .then(() => stop.execute())
```
Comment (jgraff2) [inline test/commands/stop.js]: It should be stop.execute(true) if this is testing force stop, right?
And is there a way to actually test that the queues aren't flushed?

File: test/commands/stop.js [hunk 5]
```diff
@@ -11,22 +11,183 @@
...
+    start.execute(collectorName, refocusUrl, accessToken, {})
+    .then(() => {
+      // make sure start created the repeat
+      expect(repeater.tracker.heartbeat).to.be.an('object');
+    })
+    .then(() => stop.execute())
+    .then((res) => {
+      expect(res.status).to.equal(httpStatus.OK);
+      expect(repeater.tracker.heartbeat).to.equal(undefined);
+      repeater.stopAllRepeat();
```
Comment (jgraff2) [inline test/commands/stop.js]: Is this supposed to be here?

File: test/commands/stop.js (hunk 4 above)
Comment (shriramshankar) [reply test/commands/stop.js]: yes, will fix it.

File: test/commands/stop.js (hunk 5 above)
Comment (shriramshankar) [reply test/commands/stop.js]: Nice catch. this should not be here.

**Comment by shriramshankar** [comment]
@jgraff2 @iamigo Updated the tests in the PR.

**Comment by shriramshankar** [comment]
@iamigo @jgraff2 While this PR implements what has been described in the story; I think we have to re-think about what our commands should be doing and how it should be working. We can talk more about this tomorrow.

File: src/commands/start.js [hunk 6]
```diff
@@ -58,7 +58,7 @@ function execute(collectorName, refocusUrl, accessToken, rcProxy) {
      * logging once we have heartbeat
      */
     repeater.create({
-      name: 'Heartbeat',
+      name: 'heartbeat',
```
Comment (iamigo) [inline src/commands/start.js]: move this hard-coded constant to the constants file?

Comment (shriramshankar) [reply src/commands/start.js]: @iamigo Can you merge this? I will do this in #112

**Pull request merged**